pkg/sync/syncWorker.go

```diff
@@ -441,16 +442,30 @@ func (s *syncWorker) insertEnvelope(env *envUtils.OriginatorEnvelope) {
 		return
 	}
 
-	q := queries.New(s.store)
-	inserted, err := q.InsertGatewayEnvelope(
+	originatorID := int32(env.OriginatorNodeID())
+	originatorTime := utils.NsToDate(env.OriginatorNs())
+
+	inserted, err := db.InsertGatewayEnvelopeAndIncrementUnsettledUsage(
 		s.ctx,
+		s.store,
 		queries.InsertGatewayEnvelopeParams{
 			OriginatorNodeID:     int32(env.OriginatorNodeID()),
 			OriginatorSequenceID: int64(env.OriginatorSequenceID()),
 			Topic:                env.TargetTopic().Bytes(),
 			OriginatorEnvelope:   originatorBytes,
 			PayerID:              db.NullInt32(payerId),
 		},
+		queries.IncrementUnsettledUsageParams{
+			PayerID:           payerId,
+			OriginatorID:      originatorID,
+			MinutesSinceEpoch: utils.MinutesSinceEpoch(originatorTime),
+			// TODO:(nm) Independently calculate fees
```

Comment:
This is obviously wrong. Each node should independently verify the fees instead of trusting the originator. But we will need to wait until we have a consistent payload length to do that, which requires proto updates.